### ISSUE TITLE
29.97 Non-Drop

### ISSUE THREAD

**Issue by iluvcapra** [eoyilmaz/timecode#13]
What's the supported way of creating an object that's 29.97 Non-Drop? Or 30 Drop? The `__init__` seems to automatically infer drop-frame state from the frame rate.

I could do this:

```
tc1 = Timecode('29.97','01:03:18:00')
tc1.drop_frame = False
```

Just something to keep in mind. Most feature films in the US, if they're using 29.97 code on set-- which isn't too uncommon on the sound side-- it's going to be non-drop.

**Comment by iluvcapra** [comment]
Also the code seems to default to 59.94 being drop-frame, which I think is a mistake, there's no defined frame-skipping algo for 60 frame code that I'm aware of.

**Comment by steferg** [comment]
> Also the code seems to default to 59.94 being drop-frame, which I think is a mistake, there's no defined frame-skipping algo for 60 frame code that I'm aware of.

Your assumption is incorrect.
Per SMPTE 12M-2008 (5.2.2) drop-frame can be used when dealing with 59.94 frame rates if you do not want your time code to drift vs. real time

**Comment by tymcode** [comment]
+1 for 30 nondrop support.  This could be indicated through the last delimiter; if it's a colon it should be treated as nondrop, otherwise (typically a semicolon, sometimes a period) it's drop frame.  Popular libraries like MediaInfo distinguish formats with the delimiters this way.

In 59.94 DF, it simply skips frame numbers 0, 1, 2, and 3 on the minute marks (except every tenth minute) versus dropping 0 and 1 as in 29.97DF.

Support for the format in VTRs is poor; even a lot of 720p60 HD decks output timecode as 29.97 DF and somehow everyone just accepts that every other frame has no unique timecode. But among animation and video editing and effects apps it's commonly supported.

**Comment by iluvcapra** [comment]
> What's the supported way of creating an object that's 29.97 Non-Drop? Or 30 Drop?

Any thought on this, is this a thing?

**Comment by kingb** [comment]
Heads up @jeff-vincent and myself are taking a crack at implementing this. It should also work for 59.94 NDF. 

We wrote some test cases, but it probably could use more. We're about to put it in production use, so we'll have more evidence of correctness shortly.

Expect a pull-request for review in just a moment.

**Comment by eoyilmaz** [comment]
Fixed with #23
